From c38dd0857236fea88797508872fcefa9ef65d535 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kamil=20Go=C5=82da?= <kamil.golda@protonmail.com>
Date: Wed, 22 Jan 2025 17:25:08 +0100
Subject: [PATCH 19/20] arm64: dts: qcom: sm6125-xiaomi-laurel-sprout: Correct
 reserved memory regions
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Adjust reserved memory regions to match downstream and prevent freezes.

Signed-off-by: Kamil Gołda <kamil.golda@protonmail.com>
---
 .../dts/qcom/sm6125-xiaomi-laurel-sprout.dts  | 34 +++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts b/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
index 8b62625..be22e98 100644
--- a/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
+++ b/arch/arm64/boot/dts/qcom/sm6125-xiaomi-laurel-sprout.dts
@@ -13,6 +13,13 @@
 #include "pm6125.dtsi"
 #include "pmi632.dtsi"
 
+/delete-node/ &reserved_mem1;
+/delete-node/ &adsp_pil_mem;
+/delete-node/ &ipa_fw_mem;
+/delete-node/ &ipa_gsi_mem;
+/delete-node/ &gpu_mem;
+
+
 / {
 	model = "Xiaomi Mi A3";
 	compatible = "xiaomi,laurel-sprout", "qcom,sm6125";
@@ -60,6 +67,33 @@ cmdline_mem: memory@ffd00000 {
 			reg = <0x0 0xffd40000 0x0 0x1000>;
 			no-map;
 		};
+
+		reserved_mem1: memory@46200000 {
+			reg = <0x0 0x46200000 0x0 0x4900000>;
+			no-map;
+		};
+
+		adsp_pil_mem: memory@55300000 {
+			reg = <0x0 0x55300000 0x0 0x2200000>;
+			no-map;
+		};
+
+		ipa_fw_mem: memory@57500000 {
+			reg = <0x0 0x57500000 0x0 0x10000>;
+			no-map;
+		};
+
+		ipa_gsi_mem: memory@57510000 {
+			reg = <0x0 0x57510000 0x0 0x5000>;
+			no-map;
+		};
+
+		gpu_mem: memory@57515000 {
+			reg = <0x0 0x57515000 0x0 0x2000>;
+			no-map;
+		};
+
+
 	};
 
 	gpio-keys {
-- 
2.56.0

